internal/blocksync/reactor.go

```diff
@@ -460,7 +460,7 @@ FOR_LOOP:
 			// first.Hash() doesn't verify the tx contents, so MakePartSet() is
 			// currently necessary.
 			// TODO(sergio): Should we also validate against the extended commit?
-			err = state.Validators.VerifyCommitLight(
+			err = state.Validators.VerifyCommitLightAllSignatures(
```

Comment:
(Being pedantic) I actually don't think this is right. This line is using the received second block's commit, whose voteset isn't finalized.

Agreed that every other usage is correct (and important to fix)!

So it should suffice for correctness here that 2/3rds of signatures are valid.

However for reasons noted in #1830 , I think the new change is better, and we refactor the logic to have the next two calls not verify these signatures.

Reply:
@ValarDragon Not pedantic at all. Spot on, I'd say.

The reason why we are using `VerifyCommitLightAllSignatures` here is that we decided to leave the previous behavior (i.e., skipping signatures) only in the light client logic, for the rest we took a conservative approach (bear in mind that the raison d'être of this PR was that skipping signatures was problematic when dealing with evidence verification).

However, after careful consideration of your proposal, we realize that the usage of `VerifyCommitLight*` in blocksync can be seen as a light client-ish verification as well, in that it does not produce or verify any evidence of misbehavior. So we'll open a PR to change this shortly. Bear with us.

Reply:
#1858